skorch/callbacks/training.py

```diff
@@ -401,13 +418,26 @@ def on_epoch_end(self, net, **kwargs):
         else:
             self.misses_ = 0
             self.dynamic_threshold_ = self._calc_new_threshold(current_score)
+            self.best_epoch_ = net.history[-1, "epoch"]
+            if self.load_best:
+                self.best_model_weights_ = deepcopy(net.module_.state_dict())
         if self.misses_ == self.patience:
             if net.verbose:
                 self._sink("Stopping since {} has not improved in the last "
                            "{} epochs.".format(self.monitor, self.patience),
                            verbose=net.verbose)
             raise KeyboardInterrupt
 
+    def on_train_end(self, net, **kwargs):
+        if (
+            self.load_best and self.best_epoch_ != net.history[-1, "epoch"] and
+            self.best_model_weights_ is not None
```

Comment:
```suggestion
            self.load_best and (self.best_epoch_ != net.history[-1, "epoch"])
            and (self.best_model_weights_ is not None)
```

This is probably not necessary but I'm always a bit skittish when it comes to precedence in Python.